src/models/node-dns-resolver.js

```diff
@@ -19,7 +19,7 @@ export class NodeDnsResolver {
         case 'ENODATA':
           return [];
       }
-      console.log('error', error);
+      throw new Error(error);
```

Comment:
`err` is already an instance of `Error` so you should not create a new `Error` object around it.

Note that, when throwing an `Error`, the `Error#message` is printed to the console. Here, the raw error coming from `node:dns` may be cryptic for end user. So, maybe you want to tell why this error occurs:

```suggestion
      throw new Error(`Could not resolve DNS for ${hostname}. Caused by: ${error.message}`);
```

Reply:
I rushed my modifications here :see_no_evil:, nice suggestion, I've applied it :+1: